### DISCUSSION TITLE
How to jump to a route in native JS?

### DISCUSSION THREAD

**Discussion by liu-jin-yi** [TanStack/router#1743] [Q&A] [answered]
I want to jump to the login page after the token expires in axios. I used window.location.href = "/login"; before, but there is a problem in electron. What is the correct way to jump in native JS?

**Comment by schiller-manuel** [comment]
are you looking for `redirect`? 

https://tanstack.com/router/latest/docs/framework/react/api/router/redirectFunction

**Reply by liu-jin-yi** [reply]
I need to redirect in the interceptors response in axios.

**Reply by schiller-manuel** [reply]
so `router.navigate()` ?

https://tanstack.com/router/latest/docs/framework/react/api/router/RouterType#navigate-method

**Reply by liu-jin-yi** [reply, marked as answer]
Finally, I had to extract createRouter into a new file and export it for use.
